File: leetcode/026.remove-duplicates-from-sorted-array.py
#!/usr/bin/env python3
'''
leetcode #26 remove-duplicates-from-sorted-array 删除排序数组中的重复项
https://leetcode-cn.com/problems/remove-duplicates-from-sorted-array/description/
'''
class Solution:
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        l = len(nums)
        if l <= 1:
            return l
        pos = 1
        for i in range(1, l):
            if nums[i] == nums[i - 1]:
                continue
            nums[pos] = nums[i]
            pos += 1
        return pos

File: leetcode/026.remove-duplicates-from-sorted-array.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        if nums.len() <= 1 {
            return nums.len() as i32;
        }
        let mut p1 = 1 as usize;
        let mut p2 = 1 as usize;
        let mut cur = nums[0];
        while p2 < nums.len() {
            if nums[p2] != cur {
                nums[p1] = nums[p2];
                p1 += 1;
            }
            cur = nums[p2];
            p2 += 1;
        }
        return p1 as i32;
    }
}
